ldm/generate.py

```diff
@@ -959,6 +970,47 @@ def _set_sampler(self):
 
         print(msg)
 
+    def _set_scheduler(self):
+        default = self.model.scheduler
+
+        higher_order_samplers = [
+            'k_dpm_2',
+            'k_dpm_2_a',
+            'k_heun',
+            'plms',   # Its first step is like Heun
+        ]
+        scheduler_map = dict(
+            ddim=DDIMScheduler,
+            ipndm=IPNDMScheduler,
+            k_euler=EulerDiscreteScheduler,
+            k_euler_a=EulerAncestralDiscreteScheduler,
+            k_lms=LMSDiscreteScheduler,
+            pndm=PNDMScheduler,
+        )
+
+        if self.sampler_name in scheduler_map:
+            sampler_class = scheduler_map[self.sampler_name]
+            msg = f'>> Setting Sampler to {self.sampler_name} ({sampler_class.__name__})'
+            self.sampler = sampler_class.from_config(
```

Comment:
I'd recommend to use the `extract_from_init` class here instead as this way we don't even have to load a file. You could take a look at how it's done here: https://github.com/huggingface/diffusers/pull/1247/files#r1020793115

Overall, we've been thinking just about such a method and will merge it to `main` probably early next week: https://github.com/huggingface/diffusers/pull/1247 . 

We could also make a release end of next week so that you have `set_scheduler` in a stable diffusers version.